test/Passes/pass-duplicate-fill.mlir

```diff
@@ -0,0 +1,203 @@
+  %1 = linalg.fill ins(%cst_2 : f32) outs(%0 : tensor<32x64xf32>) -> tensor<32x64xf32>
+  // CHECK: linalg.fill
+  // CHECK-NEXT: linalg.generic
+  %3 = linalg.generic {
+    indexing_maps = [#map, #map1, #map2], 
+    iterator_types = ["parallel", "reduction", "parallel"]} 
+    ins(%arg0, %arg1 : tensor<32x512xf32>, tensor<512x64xf32>) outs(%1 : tensor<32x64xf32>) {
+      ^bb0(%in: f32, %in_5: f32, %out: f32):
+        %9 = arith.mulf %in, %in_5 : f32
+        %10 = arith.addf %out, %9 : f32
+        linalg.yield %10 : f32
+  } -> tensor<32x64xf32>
+  // CHECK-NOT: linalg.fill
+  // CHECK: linalg.generic
+  %4 = linalg.generic {
+    indexing_maps = [#map, #map1, #map2],
+    iterator_types = ["parallel", "reduction", "parallel"]}
+    ins(%arg0, %arg1 : tensor<32x512xf32>, tensor<512x64xf32>) outs(%1 : tensor<32x64xf32>) {
+      ^bb0(%in: f32, %in_5: f32, %out: f32):
+        %9 = arith.mulf %in, %in_5 : f32
+        %10 = arith.addf %out, %9 : f32
+        linalg.yield %10 : f32
+  } -> tensor<32x64xf32> 
+  // CHECK-NOT: linalg.fill
+  %5 = linalg.add ins(%3, %4 : tensor<32x64xf32>, tensor<32x64xf32>) 
+                  outs(%1 : tensor<32x64xf32>) -> tensor<32x64xf32>
+  return %5 : tensor<32x64xf32> 
+}
+
+// -----
+
+func.func @matmuls(%arg0: tensor<32x32xf32>, %arg1: tensor<32x32xf32>) -> tensor<32x32xf32> {
+  // BUFF-COUNT-2: memref.alloc
+  // BUFFNOTDUP-COUNT-2: memref.alloc
+  %0 = tensor.empty() : tensor<32x32xf32>
+  %cst = arith.constant 0.0 : f32
+  %1 = linalg.fill ins(%cst : f32) outs(%0 : tensor<32x32xf32>) -> tensor<32x32xf32>
+  %2 = linalg.matmul ins(%arg0, %arg1 : tensor<32x32xf32>, tensor<32x32xf32>) 
+                     outs(%1 : tensor<32x32xf32>) -> tensor<32x32xf32>
+  %3 = linalg.matmul ins(%arg0, %2 : tensor<32x32xf32>, tensor<32x32xf32>)
+                     outs(%1 : tensor<32x32xf32>) -> tensor<32x32xf32>
+  return %3 : tensor<32x32xf32>
+}
+
+// CHECK-LABEL: matmuls
+// CHECK-SAME: %[[ARG0:.+]]: tensor<32x32xf32>, %[[ARG1:.+]]: tensor<32x32xf32>
+// CHECK: %[[EMPTY:.+]] = tensor.empty() : tensor<32x32xf32>
+// CHECK: %[[FILL:.+]] = linalg.fill ins(%{{.+}} : f32) outs(%[[EMPTY]] : tensor<32x32xf32>) -> tensor<32x32xf32>
+// CHECK: %[[MUL:.+]] = linalg.matmul ins(%[[ARG0]], %[[ARG1]] : tensor<32x32xf32>, tensor<32x32xf32>) 
+// CHECK-SAME:  outs(%[[FILL]] : tensor<32x32xf32>) -> tensor<32x32xf32>
+// CHECK: %[[FILL_1:.+]] = linalg.fill ins(%{{.+}} : f32) outs(%[[EMPTY]] : tensor<32x32xf32>) -> tensor<32x32xf32>
+// CHECK: %{{.+}} = linalg.matmul ins(%[[ARG0]], %[[MUL]] : tensor<32x32xf32>, tensor<32x32xf32>) 
+// CHECK-SAME:  outs(%[[FILL_1]] : tensor<32x32xf32>) -> tensor<32x32xf32>
+
+// -----
+
+// CHECK-LABEL: matmuls_1
+func.func @matmuls_1(%arg0: tensor<32x32xf32>, %arg1: tensor<32x32xf32>) -> tensor<32x32xf32> {
+  // BUFF-COUNT-3: memref.alloc
+  // BUFF-COUNT-1: memref.copy
+  // BUFFNOTDUP-COUNT-3: memref.alloc
+  // BUFFNOTDUP-COUNT-1: memref.copy
+  // CHECK-COUNT-2: linalg.fill
```

Comment:
There should be a copy, too, right? For the `add`?

Reply:
yes, there is 1 memref.copy

Reply:
then just add a `CHECK-COUNT-1: memref.copy` to make it clear and it LGTM. Thanks!

Reply:
the CHECK line does not bufferize, it only applies the transformation at the tensor level. Copy and allocations are checked with BUFFXX